lib/mcwamp.cpp

```diff
@@ -357,7 +357,7 @@ void BuildProgram(KalmarQueue* pQueue) {
 
 // used in parallel_for_each.h
 void *CreateKernel(std::string s, KalmarQueue* pQueue) {
-  return pQueue->getDev()->CreateKernel(s.c_str());
+  return pQueue->getDev()->CreateKernel(s.c_str(), pQueue);
```

Comment:
`CreateKernel` should probably be splitted into 2 functions:

1- the queue object should have a member function `CreateDispatch(s.c_str())` that creates and returns a dispatch
2- In `CreateDispatch`, it would call `getDev()->CreateKernel(s.c_str())`, which will only be responsible for loading a kernel from object.  

This line would be simplified into `pQueue->CreateDispatch(s.c_str())`